vb-fe/src/layout/Layout.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+
+import { Box, Flex } from "@chakra-ui/react";
+
+import { MIN_SHEET_HEIGHT } from "util/Variables";
+
+interface GenericLayoutProps {
+  children?: ReactElement | ReactElement[];
+  className?: string;
+  flex?: string;
+  show?: boolean;
+}
+
+interface LayoutWrapperProps extends Omit<GenericLayoutProps, "children"> {
+  children?:
+    | ReactElement<typeof LayoutHeader | typeof LayoutBody | typeof LayoutFooterOverlay | typeof LayoutFooter>
+    | Array<ReactElement<typeof LayoutHeader | typeof LayoutBody | typeof LayoutFooterOverlay | typeof LayoutFooter>>;
+}
+
+interface LayoutBodyProps extends Omit<GenericLayoutProps, "children"> {
+  children?:
+    | ReactElement<typeof LayoutSidebar | typeof LayoutContent>
+    | Array<ReactElement<typeof LayoutSidebar | typeof LayoutContent>>;
+}
+
+export const LayoutFooterOverlay: FunctionComponent<GenericLayoutProps> = (): JSX.Element => {
+  useEffect(() => {
+    document.body.style.overflow = "hidden";
+  }, []);
+
+  useEffect(() => {
+    return () => {
```

Comment:
Was this one needing to be outside of the other useEffect?  I see there's no dependency array so I'm curious

Reply:
so this return takes place when the component is destroyed. its when the component is dismounted. i'll add comments for this file to make things clearer